.gitignore

```diff
@@ -2,3 +2,6 @@
 .cache/
 .coverage
 doc/_build/
+*~
+*.swp
```

Comment:
FYI: `set noswapfile` in your .vimrc makes these never occur, they are so annoying

Reply:
That also makes it so vim can't recover if it crashes for whatever reason (e.g., you're using vim in an ssh session, and you lose your connection).  I have never seen vim create a backup file (`*~`) in my ~10 years of using it, so I just spent a few minutes trying to figure out _why_, and I couldn't.  Clearly vim can create them in some circumstances, but something in my vim setup must prevent it somehow.  For some discussion of this, see, e.g., [here](https://stackoverflow.com/a/750692/1467533) and [here](https://stackoverflow.com/a/607475/1467533).

But, more applicable to this change in this PR, these lines are specific to your development environment, and you should not pollute the shared `.gitignore` with lines specific to your IDE.  These should go in a [global `.gitignore` file](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).

Reply:
Noted. I agree that adding these to a global `.gitignore` is the right thing to do.